App crashes when accessing note and go back to list

It is very hard to reproduce this issue. Appears with current master and v0.10.0.

Steps:

1. Open any note from the list view
2. Close this note as quick as possible (using native back-button or top left arrow button does not matter)
3. Repeat Steps 1 - 2 until the app crashes (10 - 15 times at my device)

CC @korelstar 

```
03-31 08:55:11.909 2691-2725/? D/SyncTask:    Process Remote Note: R20  (*)  / MarkDown Test (2017-03-22 20:08:07 / null)
03-31 08:55:11.910 2691-2725/? D/SyncTask:    ... found -> Update
                                           
                                           --------- beginning of crash
03-31 08:55:11.915 2691-2725/? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #3
                                                 Process: it.niedermann.owncloud.notes, PID: 2691
                                                 java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                     at android.os.AsyncTask$3.done(AsyncTask.java:325)
                                                     at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                     at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                     at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
                                                     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                     at java.lang.Thread.run(Thread.java:761)
                                                  Caused by: java.lang.IllegalStateException: attempt to re-open an already-closed object: SQLiteDatabase: /data/user/0/it.niedermann.owncloud.notes/databases/OWNCLOUD_NOTES
                                                     at android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:55)
                                                     at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1548)
                                                     at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1525)
                                                     at it.niedermann.owncloud.notes.persistence.NoteSQLiteOpenHelper.updateNote(NoteSQLiteOpenHelper.java:407)
                                                     at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pullRemoteChanges(NoteServerSyncHelper.java:278)
                                                     at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:201)
                                                     at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:170)
                                                     at android.os.AsyncTask$2.call(AsyncTask.java:305)
                                                     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
                                                     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
                                                     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
                                                     at java.lang.Thread.run(Thread.java:761) 
```

--- a/app/src/main/java/it/niedermann/owncloud/notes/persistence/NoteSQLiteOpenHelper.java
+++ b/app/src/main/java/it/niedermann/owncloud/notes/persistence/NoteSQLiteOpenHelper.java
@@ -199,7 +199,6 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
         values.put(key_category, note.getCategory());
         values.put(key_etag, note.getEtag());
         long id = db.insert(table_notes, null, values);
-        db.close();
         return id;
     }
 
@@ -313,7 +312,6 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
         values.put(key_status, note.getStatus().getTitle());
         values.put(key_favorite, note.isFavorite()?"1":"0");
         db.update(table_notes, values, key_id + " = ?", new String[]{String.valueOf(note.getId())});
-        db.close();
         if(callback!=null) {
             serverSyncHelper.addCallbackPush(callback);
         }
@@ -344,7 +342,6 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
         values.put(key_modified, newNote.getModified(DATE_FORMAT));
         values.put(key_content, newNote.getContent());
         int rows = db.update(table_notes, values, key_id + " = ? AND " + key_content + " != ?", new String[]{String.valueOf(newNote.getId()), newNote.getContent()});
-        db.close();
         // if data was changed, set new status and schedule sync (with callback); otherwise invoke callback directly.
         if(rows > 0) {
             if(callback!=null) {
@@ -405,7 +402,6 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
             whereArgs = new String[]{String.valueOf(id), DBStatus.VOID.getTitle(), remoteNote.getModified(DATE_FORMAT), remoteNote.getTitle(), remoteNote.isFavorite()?"1":"0", remoteNote.getCategory(), remoteNote.getEtag(), remoteNote.getContent()};
         }
         int i = db.update(table_notes, values, whereClause, whereArgs);
-        db.close();
         Log.d(getClass().getSimpleName(), "updateNote: "+remoteNote+" || forceUnchangedDBNoteState: "+forceUnchangedDBNoteState+"  => "+i+" rows updated");
         return i;
     }
@@ -426,7 +422,6 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
                 values,
                 key_id + " = ?",
                 new String[]{String.valueOf(id)});
-        db.close();
         getNoteServerSyncHelper().scheduleSync(true);
         return i;
     }
@@ -443,6 +438,5 @@ public class NoteSQLiteOpenHelper extends SQLiteOpenHelper {
         db.delete(table_notes,
                 key_id + " = ? AND " + key_status + " = ?",
                 new String[]{String.valueOf(id), forceDBStatus.getTitle()});
-        db.close();
     }
 }

